src/types.rs

```diff
@@ -244,19 +244,14 @@ pub struct DateTimeSpec {
 
 // same as chrono's 'count days from monday' convention
 pub fn week_day(s: &str) -> Option<u8> {
-    let mut s = match s.as_bytes() {
-        [a, b, c, ..] => [*a, *b, *c],
-        _ => return None,
-    };
-    s.make_ascii_lowercase();
-    match &s {
-        b"sun" => Some(6),
-        b"mon" => Some(0),
-        b"tue" => Some(1),
-        b"wed" => Some(2),
-        b"thu" => Some(3),
-        b"fri" => Some(4),
-        b"sat" => Some(5),
+    match s.to_ascii_lowercase().as_str() {
```

Comment:
Using `to_ascii_lowercase` here implicitly allocates and is no longer no_std compatible

Reply:
Should be fixed now. Although I did change the implementation quite a bit.